ClosedXML/Excel/XLWorksheet.cs

```diff
@@ -1305,69 +1311,36 @@ private void ShiftPageBreaksRows(XLRange range, int rowsShifted)
 
         private void ShiftConditionalFormattingRows(XLRange range, int rowsShifted)
         {
+            if (!ConditionalFormats.Any()) return;
             Int32 firstRow = range.RangeAddress.FirstAddress.RowNumber;
             if (firstRow == 1) return;
 
-            SuspendEvents();
-            IXLRangeAddress usedAddress;
-            using (var rangeUsed = range.Worksheet.RangeUsed(true))
-            {
-                usedAddress = rangeUsed == null ? range.RangeAddress : rangeUsed.RangeAddress;
-            }
-            ResumeEvents();
-
-            if (firstRow < usedAddress.FirstAddress.RowNumber) firstRow = usedAddress.FirstAddress.RowNumber;
-
-            Int32 lastRow = range.RangeAddress.FirstAddress.RowNumber + rowsShifted - 1;
-            if (lastRow > usedAddress.LastAddress.RowNumber) lastRow = usedAddress.LastAddress.RowNumber;
-
-            Int32 firstColumn = range.RangeAddress.FirstAddress.ColumnNumber;
-            if (firstColumn < usedAddress.FirstAddress.ColumnNumber) firstColumn = usedAddress.FirstAddress.ColumnNumber;
+            int rowNum = rowsShifted > 0 ? firstRow - 1 : firstRow;
+            var model = Row(rowNum).AsRange();
 
-            Int32 lastColumn = range.RangeAddress.LastAddress.ColumnNumber;
-            if (lastColumn > usedAddress.LastAddress.ColumnNumber) lastColumn = usedAddress.LastAddress.ColumnNumber;
-
-            var insertedRange = Range(firstRow, firstColumn, lastRow, lastColumn);
-            var fr = insertedRange.FirstRow();
-            var model = fr.RowAbove();
-            Int32 modelFirstColumn = model.RangeAddress.FirstAddress.ColumnNumber;
-            if (ConditionalFormats.Any(cf => cf.Range.Intersects(model)))
+            foreach (var cf in ConditionalFormats.ToList())
             {
-                for (Int32 co = firstColumn; co <= lastColumn; co++)
+                var cfAddress = cf.Range.RangeAddress;
+                if (cf.Range.Intersects(model))
                 {
-                    using (var cellModel = model.Cell(co - modelFirstColumn + 1).AsRange()) 
-                        foreach (var cf in ConditionalFormats.Where(cf => cf.Range.Intersects(cellModel)).ToList())
-                        {
-                            using (var r = Range(firstRow, co, lastRow, co)) r.AddConditionalFormat(cf);
-                        }
+                    cf.Range = Range(cfAddress.FirstAddress.RowNumber,
+                                     cfAddress.FirstAddress.ColumnNumber,
+                                     cfAddress.LastAddress.RowNumber + rowsShifted,
+                                     cfAddress.LastAddress.ColumnNumber);
                 }
-            }
-            insertedRange.Dispose();
-            model.Dispose();
-            fr.Dispose();
-        }
-
-        internal void BreakConditionalFormatsIntoCells(List<IXLAddress> addresses)
```

Comment:
Are you sure we won't need this for something else?

Reply:
This method was not used anywhere else, so I assumed there would be no use in dead code, not even covered by tests. Besides, as a user I always hate when Excel splits a single CF into small pieces - say, from A:A it can create A1:A4, A5:A5, A6:A1048456.
I really do not see why it might be needed to split a CF into separate cells.

Reply:
Ok, there are some other PR's related to conditional formatting, but it doesn't look like they're dependent on this method.